src/tiered/tiered_handle.c

```diff
@@ -129,6 +129,18 @@ __tiered_init_tiers(WT_SESSION_IMPL *session, WT_TIERED *tiered, WT_CONFIG_ITEM
     WT_DECL_ITEM(tmp);
     WT_DECL_RET;
     WT_TIERED_TIERS *local_tier;
+    uint32_t object_id;
+    const char *cfg, *name;
+
+    if (session->import_list != NULL) {
+        /* Insert metadata for all tiered objects. */
+        for (object_id = 1; object_id < tiered->current_id; object_id++) {
+            WT_RET(
+              __wt_tiered_name(session, &tiered->iface, object_id, WT_TIERED_NAME_OBJECT, &name));
+            if ((cfg = __wt_find_import_metadata(session, name)) != NULL)
```

Comment:
Should this be an error if we can't find the metadata?

Reply:
That's a good question. @sueloverso should we fail if there's not enough information in `WiredTiger.export`?

Reply:
Thank you! The code now fails with `WT_NOTFOUND` and `failed to find metadata for <URI>` message if metadata is not found for the given URI: 49bb3672724dd407e4904e77238612af64a28108.